plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/remap/RemappingInstrumentable.kt

```diff
@@ -0,0 +1,35 @@
+@file:Suppress("UnstableApiUsage")
+
+package io.sentry.android.gradle.instrumentation.remap
+
+import com.android.build.api.instrumentation.ClassContext
+import io.sentry.android.gradle.instrumentation.ClassInstrumentable
+import io.sentry.android.gradle.instrumentation.SpanAddingClassVisitorFactory
+import org.objectweb.asm.ClassVisitor
+import org.objectweb.asm.commons.ClassRemapper
+import org.objectweb.asm.commons.SimpleRemapper
+
+class RemappingInstrumentable : ClassInstrumentable {
+    companion object {
+        private val mapping = mapOf(
+            "java/io/FileReader" to "io/sentry/instrumentation/file/SentryFileReader",
+            "java/io/FileWriter" to "io/sentry/instrumentation/file/SentryFileWriter"
+        )
+    }
+
+    override fun getVisitor(
+        instrumentableContext: ClassContext,
+        apiVersion: Int,
+        originalVisitor: ClassVisitor,
+        parameters: SpanAddingClassVisitorFactory.SpanAddingParameters
+    ): ClassVisitor =
+        ClassRemapper(originalVisitor, SimpleRemapper(mapping))
+
+    override fun isInstrumentable(data: ClassContext): Boolean {
+        return when {
+            data.currentClassData.className.startsWith("io.sentry") &&
+                !data.currentClassData.className.startsWith("io.sentry.android.roomsample") -> false
```

Comment:
also consider `io.sentry.samples.` and `io.sentry.mobile.`, similar to `SentryStackTraceFactory` on Java SDK

Reply:
`RemappingInstrumentable` should also be fixed

Reply:
Similar to https://github.com/getsentry/sentry-android-gradle-plugin/pull/239#discussion_r757341503 which was resolved but not worked on.
This is not only for samples actually since the Sentry Mobile app has a package called `io.sentry.mobile...` and it'd not work at all.

Reply:
yea, but we don't use SAGP in those, so why bother?

Reply:
👍 sounds good then